sgqlc/types/__init__.py

```diff
@@ -1337,6 +1338,23 @@ def __to_graphql__(cls, indent=0, indent_string='  '):
         suffix = ' = ' + ' | '.join(str(c) for c in cls.__types__)
         return BaseMeta.__to_graphql__(cls, indent, indent_string) + suffix
 
+    def __getitem__(cls, key):
+        return cls.__meta_fields[key]
+
+    @property
+    def __meta_fields(cls):
```

Comment:
could you follow with the same code snippet as with `ContainerTypeMeta`, with `__populate_meta_fields()`? Maybe make it a shared metaclass on top of `BaseMeta` where both `ContainerTypeMeta` and `UnionMeta` inherit, providing that meta fields